src/utils/analytics.ts

```diff
@@ -0,0 +1,23 @@
+import type { PlaylistItem, Source } from '#types/playlist';
+
+export const attachAnalyticsParams = (item: PlaylistItem, isJwIntegration: boolean, userId?: string, profileId?: string) => {
+  const { sources, mediaid } = item;
+
+  return sources.map((source: Source) => {
+    const url = new URL(source.file);
+
+    // Attach user_id and profile_id only for VOD and BCL SaaS Live Streams
+    const isVOD = url.href === `https://cdn.jwplayer.com/manifests/${mediaid}.m3u8`;
+    const isBCL = url.href === `https://content.jwplatform.com/live/broadcast/${mediaid}.m3u8`;
```

Comment:
Sorry for the noise, but slightly different suggestion to avoid case sensitivity w/o changing the source URL's which are case sensitive:
```suggestion
    const isVOD = url.href.toLowerCase() === `https://cdn.jwplayer.com/manifests/${mediaid.toLowerCase()}.m3u8`;
    const isBCL = url.href.toLowerCase() === `https://content.jwplatform.com/live/broadcast/${mediaid.toLowerCase()}.m3u8`;
```